rc/filetype/graphql.kak

```diff
@@ -27,22 +27,46 @@ hook -group graphql-highlight global WinSetOption filetype=graphql %{
 }
 
 
-provide-module graphql %(
+provide-module graphql %§ 
 
 # Highlighters
 # ‾‾‾‾‾‾‾‾‾‾‾‾
 
 add-highlighter shared/graphql regions
 add-highlighter shared/graphql/code default-region group
-add-highlighter shared/graphql/string region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/graphql/comment region '#' \n fill comment
+add-highlighter shared/graphql/line-description region '#' '\n' fill comment
+add-highlighter shared/graphql/block-description region '"""' '"""' fill comment
+add-highlighter shared/graphql/description region '"' '"\s*\n' fill comment
+add-highlighter shared/graphql/object region -recurse \{ [{] [}] regions
```

Comment:
I haven't checked, but won't this get confused by `{ foo(s: "-- } ") { bar } }`?

Reply:
yes you're absolutely right, i'm not sure this can be accounted for with just regexes or without supporting a additional nested string region?

luckily string literals within field arguments are pretty uncommon directly in graphql, the values are typically passed in programmatically from the host language. so while this is a problem, i think it's unlikely to surface in the wild

Reply:
> yes you're absolutely right, i'm not sure this can be accounted for with just regexs or without supporting an additional nested string region?
> 
> luckily string literals within field arguments are pretty uncommon directly in graphql, the values are typically passed in programmatically from the host language. so while this is a problem, i think it's unlikely to surface in the wild

I just hit this in the wild fwiw, and have no idea how to get around it :monocle_face: